account_move_transactionid_import/models/account_move.py

```diff
@@ -0,0 +1,74 @@
+# © 2011-2019 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+from odoo import _, fields, models
+from odoo.addons.account_move_base_import.models.account_move import \
+    ErrorTooManyPartner
+
+
+class AccountMoveCompletionRule(models.Model):
+    """Add a rule based on transaction ID"""
+
+    _inherit = "account.move.completion.rule"
+
+    function_to_call = fields.Selection(
+        selection_add=[
+            ('get_from_transaction_id_and_so',
+             'Match Sales Order using transaction ID'),
+            ('get_from_transaction_id_and_invoice',
+             'Match Invoice using transaction ID')
+        ])
+
+    def get_from_transaction_id_and_so(self, line):
+        """
+        Match the partner based on the transaction ID field of the SO.
+        Then, call the generic st_line method to complete other values.
+        In that case, we always fullfill the reference of the line with the SO
+        name.
+        :param dict st_line: read of the concerned account.bank.statement.line
+        :return:
+            A dict of value that can be passed directly to the write method of
+            the statement line or {}
+           {'partner_id': value,
+            'account_id' : value,
+            ...}
+            """
+        res = {}
+        so_obj = self.env['sale.order']
+        sales = so_obj.search([('transaction_id', '=', line.transaction_ref)])
+        if len(sales) > 1:
+            raise ErrorTooManyPartner(
+                _('Line named "%s" was matched by more than '
+                  'one partner.') % line.name)
+        if len(sales) == 1:
+            sale = sales[0]
+            res['partner_id'] = sale.partner_id.id
```

Comment:
IMO, can stay like that as well
```suggestion
            res['partner_id'] = sales[0].partner_id.id
```

Reply:
Will remove the previous line and simplify